repo2docker/buildpacks/julia/julia_project.py

```diff
@@ -118,7 +122,7 @@ def get_assemble_scripts(self):
             (
                 "${NB_USER}",
                 r"""
-                julia -e "using Pkg; Pkg.add(\"IJulia\"); using IJulia; installkernel(\"Julia\", \"--project=${REPO_DIR}\", env=Dict(\"JUPYTER_DATA_DIR\"=>\"${NB_PYTHON_PREFIX}/share/jupyter\"));" && \
+                JULIA_PROJECT="" julia -e "using Pkg; Pkg.add(\"IJulia\"); using IJulia; installkernel(\"Julia\", env=Dict(\"JUPYTER_DATA_DIR\"=>\"${NB_PYTHON_PREFIX}/share/jupyter\"));" && \
```

Comment:
For my education: Why set it to an empty value here? Not sure what I was expecting but explicitly setting it to be empty surprised me :)

Reply:
I assume this is to make sure we install the default version of IJulia? But actually, that is a bit interesting... might we not also want to allow the user to specify a different version of IJulia if they so choose?

Reply:
I wonder if we should activate and instantiate the user's project first, and then only add IJulia to the default environment if they don't already have IJulia installed? And only _then_ install the kernel?

(Note that my above comments are orthogonal to this PR)

Reply:
Yes, @NHDaly is correct: By setting this to `""` we make sure the IJulia package gets installed into the default environment, _not_ the user specified environment. Later, these environments are automatically stacked by julia. So if a user also had an IJulia version in their `Project.toml`, it would take precedence over the one we installed in the default environment.

I don't think we should install IJulia into the user provided env: that can potentially lead to updates of over, unrelated packages, and then we really break reproducability.

Reply:
> I don't think we should install IJulia into the user provided env: that can potentially lead to updates of over, unrelated packages, and then we really break reproducability.

Yeah, I had the same concern. I agree with you here 100%!

> So if a user also had an IJulia version in their Project.toml, it would take precedence over the one we installed in the default environment.

But what i was concerned about is that if a user specifies a different version of IJulia in their Project.toml, we're still going to install the _kernel_ from this default version, not the one they specify. Would it be crazy to peek ahead and only install IJulia into the default env if the user doesn't include it in theirs? (Either way, this seems like a feature that could be added in a later PR)